### ISSUE TITLE
Core code quality issues may not be noticed by code quality auditing tools

### ISSUE THREAD

**Issue by ericmorand** [magento/magento2#30446]
### Preconditions (*)

None. This issue applies to every Magento 2 release since its creation.
Magento 2.4-develop

### Steps to reproduce (*)

Execute a global search of  `@SuppressWarnings` in the packages located under the _magento_ vendor.

Using PHPStorm to perform the search, I get *8462* (eight thousand four hundred and sixty two) occurences of `@SuppressWarnings`.

### Expected result (*)

There should be zero occurence of `@SuppressWarnings` in the Core source code of Magento 2.

### Actual result (*)
There are lots of `@SuppressWarnings` in the Core source code of Magento 2.

### What does it mean?

It means that the Core maintainers are deliberately hiding potential code quality issues from the Core source code by using directives that make the code quality audit tools ignoring some issues.

For example, the `Magento\Framework\View\Result\Page` class alone is plagued by four potential quality code violations that would never be noticed by the code quality auditing tool:

```php
 * @SuppressWarnings(PHPMD.TooManyFields)
 * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
 * @SuppressWarnings(PHPMD.DepthOfInheritance)
 * @SuppressWarnings(PHPMD.ExcessiveParameterList)
```

In other words: Magento 2 code quality _may be_ a a fraud, and a deliberate one. 

I'm not saying that it _is_. Magento 2 code quality _may_ be absolutely perfect, but these directives that suppress code quality warnings cast a shadow of doubt over the honesty of the maintainers. Magento 2 Core maintainers should seriously consider removing all these `@SuppressWarnings` directives, run their code quality audit and publish the result. This is the only way to remove the doubts.

**Comment by ericmorand** [comment]
It seems like `@SupressWarnings` is not the only code quality override that is present in Magento 2.

* `@codingstandardsIgnoreFile` is used 55 times.
* `@codingstandardsIgnoreLine` is used 24 times.

**Comment by ihor-sviziev** [comment]
Hi @ericmorand,
You're totally right! Unfortunately we have quite a lot of code that was written when coding standards weren't enforced by static tests and a lot of pieces of Magento 2 have to be improved. Unfortunately fixing all these parts together isn't really possible, each specific case should be revisited and fixed.

We're welcoming you to create Pull Requests that will fix such cases in order to fix this issue.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-40839` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by ihor-sviziev** [comment]
https://github.com/magento/magento2/pull/31965 fixed codingStandardsIgnoreFile

**Comment by engcom-Hotel** [comment]
Hello @ericmorand,

We have reverified the issue in the latest 2.4-develop branch and the issue is still reproducible. We have found approx. ~7906 occurences of `@SuppressWarnings`.

Hence reconfirming the issue.

Thanks
